From 95e60867b3172eb64e3f720a3457923f70ea820c Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 1 Jul 2019 11:24:13 +0800
Subject: [PATCH] pyanaconda: payload: Fix xserver detection

When installing an image that has graphical interface,
/etc/systemd/system/default.target will point to multi-user.target
instead of graphical.target.

To fix this, use the right rpm query to detect the presence of xserver.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/payload/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyanaconda/payload/__init__.py b/pyanaconda/payload/__init__.py
index cbe2154..6b3e5ff 100644
--- a/pyanaconda/payload/__init__.py
+++ b/pyanaconda/payload/__init__.py
@@ -631,7 +631,7 @@ class Payload(metaclass=ABCMeta):
             ts = rpm.TransactionSet(util.getSysroot())
 
             # XXX one day this might need to account for anaconda's display mode
-            if ts.dbMatch("provides", 'service(graphical-login)').count() and \
+            if ts.dbMatch('provides', 'xserver-xorg').count() and \
                not flags.usevnc:
                 # We only manipulate the ksdata.  The symlink is made later
                 # during the config write out.
-- 
2.7.4

